File: 27-remove-element/27-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        if (nums.length == 0) {
            return 0;
        }

        int backIndex = nums.length - 1;
        for (int i = 0; i < backIndex; ++i) {
            if (nums[i] == val) {
                while (nums[backIndex] == val) {
                    backIndex--;
                    if (backIndex == i) {
                        return nums[backIndex] == val ? backIndex : backIndex + 1;
                    }
                }
                nums[i] = nums[backIndex];
                nums[backIndex] = val;
                backIndex--;
            }
        }
        return nums[backIndex] == val ? backIndex : backIndex + 1;
    }
}

File: 27-remove-element/27-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums) == 0 :
            return 0
        

        backIndex = len(nums) - 1
        i = 0
        while i < backIndex:
            if nums[i] == val :
                while nums[backIndex] == val:
                    backIndex -= 1
                    if backIndex == i:
                        return backIndex if nums[backIndex] == val else backIndex + 1
                    
                nums[i] = nums[backIndex]
                nums[backIndex] = val
                backIndex -= 1
            i += 1
            
        
        return backIndex if nums[backIndex] == val else backIndex + 1  